### PR TITLE
adding relaxed type-equivalency to ReParameter()

### PR DESCRIPTION
**Pull request by cmstein**
Signed-off-by: Clifford Stein <cstein@imageworks.com>

## Description

Allows ShadingSystem->ReParameter() to use the same relaxed rules as SS->Parameter()

## Tests

no changes in tests

## Checklist:

- [x] I have read the [contribution guidelines](../CONTRIBUTING.md).
- [x] I have previously submitted a [Contributor License Agreement](http://opensource.imageworks.com/cla/).
- [x] I have updated the documentation, if applicable.
- [x] I have ensured that the change is tested somewhere in the testsuite (adding new test cases if necessary).
- [x] My code follows the prevailing code style of this project.

### CODE REVIEW

File: src/include/osl_pvt.h
Comment (lgritz) [inline src/include/osl_pvt.h:407]: Let's move this above to right after 'equivalent' and declare as a friend.

**Review by lgritz — COMMENTED** [review]
LGTM modulo the one comment and it needs clang-format.

**Comment by lgritz** [comment]
@cmstein, can you please fix the clang-format (a line each in typedesc.cpp and shadingsys.cpp) and also edit the last commit you have to be sure it has the "Signed-off-by:" message for the DCO? Thanks.

I tried to do this myself, but for some reason can't seem to push it back to your branch (despite the message saying "Add more commits by pushing to the [relaxed_equiv](https://github.com/cmstein/OpenShadingLanguage/tree/relaxed_equiv) branch on [cmstein/OpenShadingLanguage](https://github.com/cmstein/OpenShadingLanguage).", which I should be able to do as the maintainer of the project. I'm not sure why this doesn't work.

**Comment by cmstein** [comment]
fixed clang-format issues and squashed the commits

**Review by lgritz — APPROVED** [review]

**Pull request merged**
